acme/order.go

```diff
@@ -289,14 +295,22 @@ func canonicalize(csr *x509.CertificateRequest) (canonicalized *x509.Certificate
 	// identifiers as the initial newOrder request. Identifiers of type "dns"
 	// MUST appear either in the commonName portion of the requested subject
 	// name or in an extensionRequest attribute [RFC2985] requesting a
-	// subjectAltName extension, or both.
-	// TODO(hs): we might want to check if the CommonName is in fact a DNS (and cannot
-	// be parsed as IP). This is related to https://github.com/smallstep/cli/pull/576
-	// (ACME IP SANS)
+	// subjectAltName extension, or both. Subject Common Names that can be
+	// parsed as an IP are included as an IP address for the equality check.
+	// If these were excluded, a certificate could contain an IP as the
+	// common name without having been challenged.
 	if csr.Subject.CommonName != "" {
-		// nolint:gocritic
-		canonicalized.DNSNames = append(csr.DNSNames, csr.Subject.CommonName)
+		ip := net.ParseIP(csr.Subject.CommonName)
+		subjectIsIP := ip != nil
```

Comment:
I wann have a quick discussion about syntax here. Historically we would write the line above as: 
```
if ip := net.ParseIP(blah); ip != nil {
   // do something
} else {
  // do something else
}
```

But you're breaking that first line up into three separate lines. I'm on board with breaking up code into digestible chunks, but I kinda feel like we're oversimplifying here and avoiding some useful shortcuts. Like, making `subjectIsIP` a variable seems kinda unnecessary if we're just going to consume that variable on the very next line.

Most importantly, I think we should just be consistent within the codebase. So if we'd prefer to use the slightly longer but more clear syntax (like in your PR), that's cool. We can slowly switch change the codebase over time, but maybe let's just have a discussion about it. If we're not going to use it everywhere, it feels kinda strange to only use it sometimes.